x-pack/plugins/apm/server/routes/suggestions/get_suggestions_with_terms_aggregation.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import moment from 'moment';
+import { ProcessorEvent } from '../../../common/processor_event';
+import { SERVICE_NAME } from '../../../common/elasticsearch_fieldnames';
+import { rangeQuery, termQuery } from '../../../../observability/server';
+import { Setup } from '../../lib/helpers/setup_request';
+
+export async function getSuggestionsWithTermsAggregation({
+  fieldName,
+  fieldValue,
+  serviceName,
+  setup,
+  size,
+  start = moment().subtract(24, 'h').valueOf(),
+  end = moment().valueOf(),
+}: {
+  fieldName: string;
+  fieldValue: string;
+  serviceName: string;
+  setup: Setup;
+  size: number;
+  start?: number;
+  end?: number;
+}) {
+  const { apmEventClient } = setup;
+
+  const response = await apmEventClient.search(
+    'get_suggestions_with_terms_aggregation',
+    {
+      apm: {
+        events: [
+          ProcessorEvent.transaction,
+          ProcessorEvent.error,
+          ProcessorEvent.metric,
+        ],
+      },
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...termQuery(SERVICE_NAME, serviceName),
+              ...rangeQuery(start, end),
+              {
+                wildcard: {
+                  [fieldName]: `*${fieldValue}*`,
```

Comment:
should we use the leading asterisk here? [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-wildcard-query.html) mention the following:

> Avoid beginning patterns with * or ?. This can increase the iterations needed to find matching terms and slow search performance.

Reply:
IMO we should use it cause _unfortunately_ we have cases where they pass a host name as environment value and without the leading asterisk the search experience will be poor. 

For example, I have this environment `https://www.elastic.co` and if I try to search for `elas` without the leading asterisk the results will be empty. I doubt the user will search for `https` to get `https://www.elastic.co`

Reply:
I've thought about this a bit, and I feel we should add a `timeout` here, something like 1500ms. These results don't need to be accurate and it's better to return possibly incomplete suggestions in a relatively short time frame rather than having the user potentially wait double-digit seconds before getting suggestions. There's more information about the `timeout` property here. https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html

Reply:
@kpatticha did you see this comment? I don't see the timeout being added.

Reply:
sorry for missing your comment @dgieselaar, it's resolved here https://github.com/elastic/kibana/pull/126679/commits/33ab81bd697d911c5807fbc5ba3cc0d26dd04874